cscs-checks/apps/paraview/paraview_check.py

```diff
@@ -9,32 +9,28 @@
 
 @rfm.simple_test
 class ParaViewCheck(rfm.RunOnlyRegressionTest):
-    def __init__(self):
-        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
-                              'eiger:mc', 'pilatus:mc']
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc',
+                     'eiger:mc', 'pilatus:mc']
+    valid_prog_environs = ['builtin']
+    num_tasks = 12
+    num_tasks_per_node = 12
+    modules = ['ParaView']
+    executable = 'pvbatch'
+    executable_opts = ['coloredSphere.py']
+    maintainers = ['JF', 'TM']
+    tags = {'scs', 'production'}
 
-        if self.current_system.name in ['eiger', 'pilatus']:
+    @run_after('setup')
```

Comment:
Here is the bug. This should be a post-init hook:

```suggestion
    @run_after('init')
```

The reason for that is that after setup is too late for test filtering; the test has already started running (you can use `skip()` there, but it's much simpler if you just change the hook.